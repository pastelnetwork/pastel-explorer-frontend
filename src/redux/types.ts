import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { Action } from 'redux';
import { RootState } from '@redux/store';

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export type AppThunkDispatch = ThunkDispatch<RootState, unknown, Action<string>>;
