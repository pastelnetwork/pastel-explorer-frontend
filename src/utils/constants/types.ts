import { LabelKeyObject } from 'react-csv/components/CommonPropTypes';
import { PeriodTypes, TGranularity } from '@utils/helpers/statisticsLib';
import { TStatisticsInfo } from '@utils/types/IStatistics';

export type TThemeColor = {
  name: string;
  backgroundColor: string;
  lineColor?: string;
  splitLineColor: string;
  color: string;
  stack?: string;
  smooth?: boolean;
};

export type TThemeInitOption = {
  theme?: TThemeColor | null;
  data?: number[][];
  dataX?: string[];
  dataY?: number[];
  dataY1?: number[];
  dataY2?: number[];
  chartName: string;
  minY: number;
  maxY: number;
};

export type TLineChartProps = {
  chartName: string;
  period?: PeriodTypes;
  dataX?: string[];
  dataY?: number[];
  dataY1?: number[];
  dataY2?: number[];
  title?: string;
  granularity?: TGranularity;
  info: TStatisticsInfo;
  granularities?: TGranularity[];
  offset: number;
  periods?: PeriodTypes[];
  handleGranularityFilterChange?: (_granularity: TGranularity) => void;
  handleBgColorChange: (_color: string) => void;
  handlePeriodFilterChange?: (_period: PeriodTypes) => void;
};

export type TCsvHeaderType = {
  [key: string]: LabelKeyObject[];
};

export type TScatterChartProps = {
  chartName: string;
  data: number[][];
  dataX: string[];
  title?: string;
  period?: PeriodTypes;
  info: TStatisticsInfo;
  offset: number;
  periods: PeriodTypes[];
  handleBgColorChange: (_color: string) => void;
  handlePeriodFilterChange?: (_period: PeriodTypes) => void;
  handleGranularityFilterChange?: (_granularity: TGranularity) => void;
};
