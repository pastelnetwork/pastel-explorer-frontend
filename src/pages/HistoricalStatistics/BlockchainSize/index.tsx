// react
import { useEffect, useState } from 'react';
// third party
import { Skeleton } from '@material-ui/lab';
// application
import * as URLS from '@utils/constants/urls';
import { useFetch } from '@utils/helpers/useFetch/useFetch';
import { PeriodTypes, transformBlockchainSize } from '@utils/helpers/statisticsLib';
import { periods, info } from '@utils/constants/statistics';
import { useBackgroundChart } from '@utils/hooks';
import { TLineChartData, TTransactionsChart } from '@utils/types/IStatistics';
import HistoricalStatisticsLayout from '@components/HistoricalStatisticsLayout';

import { EChartsLineChart } from '../Chart/EChartsLineChart';

function BlockchainSize() {
  const [chartData, setChartData] = useState<TLineChartData | null>(null);
  const [currentBgColor, handleBgColorChange] = useBackgroundChart();
  const [period, setPeriod] = useState<PeriodTypes>(periods[1][0]);
  const fetchStats = useFetch<{ data: Array<TTransactionsChart> }>({
    method: 'get',
    url: URLS.GET_BLOCKS_CHARTS,
  });
  useEffect(() => {
    const loadLineChartData = async () => {
      const data = await fetchStats.fetchData({
        params: { period, sortDirection: 'DESC', func: 'SUM', col: 'size' },
      });
      if (data) {
        const parseData = transformBlockchainSize(data.data);
        setChartData(parseData);
      }
    };
    loadLineChartData();
  }, [period]);

  const handlePeriodFilterChange = (value: PeriodTypes) => {
    setPeriod(value);
  };

  return (
    <HistoricalStatisticsLayout currentBgColor={currentBgColor}>
      {chartData ? (
        <EChartsLineChart
          chartName="transactionfee"
          dataX={chartData?.dataX}
          dataY={chartData?.dataY}
          title="Blockchain Size (Mb)"
          info={info}
          period={period}
          offset={1}
          periods={periods[1]}
          handleBgColorChange={handleBgColorChange}
          handlePeriodFilterChange={handlePeriodFilterChange}
        />
      ) : (
        <Skeleton animation="wave" variant="rect" height={386} />
      )}
    </HistoricalStatisticsLayout>
  );
}

export default BlockchainSize;
