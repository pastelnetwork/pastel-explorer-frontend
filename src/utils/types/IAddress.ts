export interface IAddressData {
  amount: number;
  timestamp: number;
  transactionHash: string;
  direction: string;
}

export interface IAddress {
  address: string;
  incomingSum: number;
  outgoingSum: number;
  data: Array<IAddressData>;
}
