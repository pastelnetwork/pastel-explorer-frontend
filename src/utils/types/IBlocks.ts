export interface IBlockTransaction {
  id: string;
  recipientCount: number;
  totalAmount: number;
}

export interface IBlock {
  confirmations: number;
  difficulty: string;
  height: number;
  id: string;
  merkleRoot: string;
  nextBlockHash: string;
  nonce: string;
  previousBlockHash: string;
  size: number;
  solution: string;
  timestamp: number;
  transactionCount: number;
  transactions: Array<IBlockTransaction>;
}

export interface IRawBlock {
  anchor: string;
  bits: string;
  chainwork: string;
  confirmations: number;
  difficulty: string;
  finalsaplingroot: string;
  hash: string;
  height: number;
  merkleroot: string;
  nonce: string;
  previousblockhash: string;
  size: number;
  solution: string;
  time: number;
  transactions: Array<IBlockTransaction>;
  tx: string[];
}
