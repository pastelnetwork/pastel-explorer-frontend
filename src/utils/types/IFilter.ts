export type TFilter = {
  name: string;
  value: string;
};
