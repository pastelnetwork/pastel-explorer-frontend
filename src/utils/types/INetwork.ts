export interface INetworkPeers {
  city: string;
  country: string;
  id: string;
  ip: string;
  latitude: number;
  longitude: number;
  nodeId: number;
  protocol: string;
  version: number;
}

export interface INetworkSupernodes {
  address: string;
  city: string;
  country: string;
  id: string;
  ip: string;
  lastPaidBlock: number;
  lastPaidTime: number;
  latitude: number;
  longitude: number;
  port: string;
  status: string;
  snPastelIdPubkey: string;
  rankAsOfBlockHeight: number;
  protocolVersion: number;
  masternodeRank: number;
  dateTimeLastSeen: number;
  activeSeconds: number;
}

export interface INetwork {
  peers: Array<INetworkPeers>;
  masternodes: Array<INetworkSupernodes>;
}
