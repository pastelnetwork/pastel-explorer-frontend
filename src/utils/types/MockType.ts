import { TFunction } from 'i18next';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type MyMockType = jest.Mock<any, any, any> &
  TFunction<['translation', ...string[]], undefined>;
