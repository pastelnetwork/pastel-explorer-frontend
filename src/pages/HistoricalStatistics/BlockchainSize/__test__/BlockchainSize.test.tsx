import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import 'jest-styled-components';

import { MyMockType } from '@utils/types/MockType';
import i18next from '../../../../utils/helpers/i18n';
import BlockchainSize from '../index';

jest.mock('i18next-http-backend');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            // noop
          }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {
      // noop
    },
  },
}));
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
}) as MyMockType;
const mockStore = configureMockStore();
const store = mockStore({});

describe('pages/HistoricalStatistics/BlockchainSize', () => {
  const wrapper = shallow(
    <Provider store={store}>
      <BlockchainSize />
    </Provider>,
  );

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
