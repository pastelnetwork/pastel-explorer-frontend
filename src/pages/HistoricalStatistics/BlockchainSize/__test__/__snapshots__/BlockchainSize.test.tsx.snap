// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/HistoricalStatistics/BlockchainSize renders correctly 1`] = `ShallowWrapper {}`;
